### ISSUE TITLE
Simple Map Creator And Server Creation ( Land Mode)

### ISSUE THREAD

**Issue by xXxToxicBlueDustxXx** [yvt/openspades#494]
It would be great if they could put a simple functionality that would make the game could be played in Lan this would make much better the development of the game to test things without entering x server and also a simple map maker so that users can create their own maps

**Comment by VelocityRa** [comment]
You can't put two completely unrelated and potentially huge undertakings in the same Issue...

**Comment by feikname** [comment]
The game can be played in LAN already. You just need to create your own local server using any of the available software for this (vanilla, psypades, pysnip) and then get the server URL.

A LAN detector entry could be added to search for servers, but at this point, it's probably easier to just tell your friend in the same network the server address.

**Comment by VelocityRa** [comment]
Exactly, the only difference is that you'd need to enter the server's _local_ IP (can be found easily with `ipconfig` (`ifconfig` for linux).

**Comment by noway421** [comment]
imaging how sick it would be to make server discovery through zeroconf, bonjour or the like using service advertisement.

but seriously don't take this as a feature request, too complicated and the use cases are so rare it's not worth the effort.

**Comment by feikname** [comment]
@yvt @VelocityRa can we add the denied tag to this?

**Comment by xXxToxicBlueDustxXx** [comment]
itno a server browser its juts a simple way to make a lan server o server whitout to goin to hosting.buildandshot.com :I

**Comment by noway421** [comment]
pysnip and OS are different projects and combining them in one package wouldn't be a good option I would think. Some external bundle might be a possibility, but outside of the official repo maybe?

As for openspades server, this one is not in developments as of now and even then it's `ng-spades` protocol only.

**Comment by NotAFile** [comment]
Would it really be a big problem to bundle a recent pysnip install with each release? There's tools that can just bundle a python program into a single executable, I think having an option to optionally ship with that could be nice. On the other hand, this would take manpower OS does not currently have and building a one-click pysnip deployer could just be a better solution.

**Comment by feikname** [comment]
@NotAFile @noway421  I rethinked my opinion after reading your comments above.

Maybe a next version (0.1.1) **could** have a GUI for server creation mode. It might take some time, but it surely is possible, given that pySnip is already programmed and doesn't receive updates for over a year, focusing for compatibility is easy.

Making different versions of OS complicates things, my suggestion is to add a "Download pySnip x.x.x" button in the server creation GUI and add basic configuration settings. Could be a custom build of pySnip made by us too, with some improvements and hacks to work better with OS (also some bug fixes).

I am not sure how troublesome that'd be.

**Comment by NotAFile** [comment]
@feikname A "Download PySnip" button is definitely an option since OS already has HTTP and Zip file support, but at just over 7mb size (although it is a significant file size increase), I don't see a reason not to just bundle it. The build release command could download and extract the latest windows zip and bundle it like the other resources. IMO, especially for LAN parties it would be ideal to just be able to start a server though, like with the original AoS.

Pysnip has a number of APIs, so I don't think manipulating STDIN would be the best way to do communicate with the process. It has SSH control (adding an ssh client to OS would be a terrible idea), IRC control (more viable) and HTTP JSON control (by far the best option, but it is readonly). I don't think any of that is necessary though, since you can do everything you can do on stdin with the ingame chat too. Just having a UI to edit the config file, or even just a "open config file in editor" button would be enough, really.

Long-term, I think there is no way around the OS team/community forking `pysnip` or working together with the devs (if they still have interest). I have seen that there has been work on an OS ng Server (I am not sure how far along this is), but I think forking pysnip is a better use of the very limited manpower OS has.

**Comment by Tapmemer** [comment]
why dont they make the pyspades system (and enet) have  a peer to peer functionality?

just set up and run

**Comment by noway421** [comment]
What do you mean by that? Shifting from client-server architecture to P2P is pretty much impossible while maintaining compatibility with existing client.

**Comment by roadapathy** [comment]
Hey guys, FYI. The Pyspades server has never once worked for me on Linux. I have no idea why. I have compiled WoW servers and installed them. I have installed nearly all of the Valve server stuff in Linux. I think a few other game servers but this one does not work. :-( We need something. Linux is the perfect OS to run it too.

**Comment by noway421** [comment]
@roadapathy 
We've been working on this one lately https://github.com/piqueserver/piqueserver which is a pyspades server with improvements and simpler build/install process (docker support too!). 

Come over and take a look at it, the installation is really easy: https://github.com/piqueserver/piqueserver/wiki/Installation
